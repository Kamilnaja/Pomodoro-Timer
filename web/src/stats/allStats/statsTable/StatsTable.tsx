import React from 'react';
import { Table } from 'react-bootstrap';
import { PomodorosDoneInDay } from '../../../../../types/statisticsInterfaces';
import { StatsTableProps } from './StatsTableProps';

export const StatsTable = (props: StatsTableProps) => {
  const daysInMonth = (month: number, year: number) => new Date(year, month, 0).getDate();

  const findPomodorosInDay = (day: number): PomodorosDoneInDay =>
    props.pomodoros.find(v => new Date(v.date).getDate() === day);

  const longArr = (
    <>
      {Array.from(Array(daysInMonth(props.pageMonth + 1, props.pageYear)), (e, i) => (
        <tr key={i}>
          <td className="table__date">{i + 1}</td>
          <td className="table__count">{findPomodorosInDay(i + 1)?.count}</td>
        </tr>
      ))}
    </>
  );

  const shortArr = (
    <>
      {props.pomodoros?.map((v, idx) => (
        <tr key={idx}>
          <td className="table__date">{v.date}</td>
          <td className="table__count">{v.count}</td>
        </tr>
      ))}
    </>
  );

  return (
    <Table className="stats__table table">
      <caption>
        Results from: {props.pageMonth + 1}.{props.pageYear}
      </caption>
      <thead className="table__head">
        <tr>
          <th>Day</th>
          <th>Number of pomodoros</th>
        </tr>
      </thead>
      <tbody>{longArr}</tbody>
    </Table>
  );
};
