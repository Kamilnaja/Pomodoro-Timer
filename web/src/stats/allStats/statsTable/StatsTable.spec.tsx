import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { createPomodorosDoneInDay } from '../../testing/stats.test.data';
import { StatsTable } from './StatsTable';

describe('StatsTable', () => {
  let wrapper = shallow(<StatsTable pageMonth={11} pageYear={2020} pomodoros={createPomodorosDoneInDay()} />);

  it('Should match snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
