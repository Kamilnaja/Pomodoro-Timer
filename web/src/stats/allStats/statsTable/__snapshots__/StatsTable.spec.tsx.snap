// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsTable Should match snapshot 1`] = `
<ForwardRef
  className="stats__table table"
>
  <caption>
    Results from: 
    12
    .
    2020
  </caption>
  <thead
    className="table__head"
  >
    <tr>
      <th>
        Day
      </th>
      <th>
        Number of pomodoros
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td
        className="table__date"
      >
        1
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="1"
    >
      <td
        className="table__date"
      >
        2
      </td>
      <td
        className="table__count"
      >
        9
      </td>
    </tr>
    <tr
      key="2"
    >
      <td
        className="table__date"
      >
        3
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="3"
    >
      <td
        className="table__date"
      >
        4
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="4"
    >
      <td
        className="table__date"
      >
        5
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="5"
    >
      <td
        className="table__date"
      >
        6
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="6"
    >
      <td
        className="table__date"
      >
        7
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="7"
    >
      <td
        className="table__date"
      >
        8
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="8"
    >
      <td
        className="table__date"
      >
        9
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="9"
    >
      <td
        className="table__date"
      >
        10
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="10"
    >
      <td
        className="table__date"
      >
        11
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="11"
    >
      <td
        className="table__date"
      >
        12
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="12"
    >
      <td
        className="table__date"
      >
        13
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="13"
    >
      <td
        className="table__date"
      >
        14
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="14"
    >
      <td
        className="table__date"
      >
        15
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="15"
    >
      <td
        className="table__date"
      >
        16
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="16"
    >
      <td
        className="table__date"
      >
        17
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="17"
    >
      <td
        className="table__date"
      >
        18
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="18"
    >
      <td
        className="table__date"
      >
        19
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="19"
    >
      <td
        className="table__date"
      >
        20
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="20"
    >
      <td
        className="table__date"
      >
        21
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="21"
    >
      <td
        className="table__date"
      >
        22
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="22"
    >
      <td
        className="table__date"
      >
        23
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="23"
    >
      <td
        className="table__date"
      >
        24
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="24"
    >
      <td
        className="table__date"
      >
        25
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="25"
    >
      <td
        className="table__date"
      >
        26
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="26"
    >
      <td
        className="table__date"
      >
        27
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="27"
    >
      <td
        className="table__date"
      >
        28
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="28"
    >
      <td
        className="table__date"
      >
        29
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="29"
    >
      <td
        className="table__date"
      >
        30
      </td>
      <td
        className="table__count"
      />
    </tr>
    <tr
      key="30"
    >
      <td
        className="table__date"
      >
        31
      </td>
      <td
        className="table__count"
      />
    </tr>
  </tbody>
</ForwardRef>
`;
